ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/customercenter/viewmodel/CustomerCenterViewModel.kt

```diff
@@ -34,31 +35,36 @@ internal interface CustomerCenterViewModel {
     fun contactSupport(context: Context, supportEmail: String)
     fun openAppStore(context: Context)
     fun showManageSubscriptions(context: Context, productId: String)
+    fun displayFeedbackSurvey(
+        feedbackSurvey: CustomerCenterConfigData.HelpPath.PathDetail.FeedbackSurvey,
+        onOptionSelected: (CustomerCenterConfigData.HelpPath.PathDetail.FeedbackSurvey.Option?) -> Unit,
+    )
```

Comment:
If we're being MVVM purists, the view should not be telling the ViewModel what to do. So this would be something like `onCancelSubscriptionClick()` (because that's the UI event the ViewModel needs to know about), and then the ViewModel determines the appropriate state, which would be to show the feedback survey. Not saying this is a blocker and we should do this, but wanted to give this perspective 🙂

Reply:
That makes sense! I had actually refactored this in https://github.com/RevenueCat/purchases-android/pull/2011/files#diff-869560546e1bbd7f1c37b444e38d8537d0d734b488ae4a1c9465978324542aa3 so bringing that into this PR for easier reviews